File: others/strings_and_chef.py
#!/usr/bin/env python3

strings = list()
counts = list()

for i in range(int(input())) :
    strings.append(input())
Q = int(input())
for i in range(Q) :
    queires = input().split()
    L = int(queires[0])
    R = int(queires[1])
    C = int(queires[2])
    P = queires[3]

    count = 0
    for string in strings[L-1:R] :
        try :
            if string[C-1] == P :
                count += 1
        except :
            continue
    counts.append(count)
for count in counts :
    print(count)

File: others/strings_and_chef.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
  int Q, N, L, R, C, i, count, j;

  i = 0;
  scanf("%d", &N);
  char *strings[N], P;
  while(i < N)
  {
      scanf("%s", strings[i]);
      i++;
  }
  scanf("%d", &Q);
  int counts[Q];
  i = 0;
  while(i < Q)
  {
      scanf("%d%d%d%c", &L, &R, &C, &P);
      count = 0;
      for(j = L-1; j < R; j++)
      {
          if((strings[j])[C-1] == P)
	  {
	      count++;
	  }
      counts[i] = count;
      }
  }
  for (i = 0; i < Q; i++)
  {
      printf("%d\n", counts[i]);
  }

  exit(0);
}
